### ISSUE TITLE
Warp tracks

### ISSUE THREAD

**Issue by 128-bit-guy** [Railcraft/Railcraft#1798]
**Is your feature request related to a problem? Please describe.**
At the moment, Railcraft's means of transport are slower than end game armor and portals from other mods. This got me thinking of warp tracks, similar to warp itemducts from Thermal dynamics for 1.7.10.

**Describe the solution you'd like**
There will be a warp track type. When powered, minecarts and trains will pass immediately from start to end of track. They will respect all turns and switches, activate detectors and detector tracks, unload and load entities on disembarking / embarking tracks, work with loaders / unloaders and track behaviours.

**Describe alternatives you've considered**
I don't know any alternatives in Railcraft. There are no alternatives in other mods either, because portals don't work with trains, detectors, loaders and unloaders.

**Additional context**
Thermal dynamics (these itemducts only exist in versions for 1.7.10):
https://www.curseforge.com/minecraft/mc-mods/thermal-dynamics/files

**Comment by bookerthegeek** [comment]
I saw this and thought to myself yes! And then I read it and thought hell no. I personally don't believe this fits the idea that is railcraft. Also you have high speed tracks.

That being said, I would like a variant of this idea but that is something for another issue submission.

**Comment by liach** [comment]
Do you want this to be cross dimensional?

Currently, Railcraft has sort of a "warp" track kit called "launcher track kit". It can make single minecarts travel in air for a distance.

**Comment by bookerthegeek** [comment]
@liach I was thinking that. I was imagining a track that when you place ot it matches to another track in the nether, similar to how a nether portal works. 

This would teleport carts or trains and a good work around to the fact that nether portals and trains are a bit buggy.

**Comment by 128-bit-guy** [comment]
No, I was thinking of track type, not kit. When minecart will appear at start, it will instantly be teleported to end of warp track line, similar to warp item ducts from thermal dynamics

**Comment by liach** [comment]
how do we know if there is a start and an end? how should signals switches and track kit functionalities be handled? the skipping here simply does not work

**Comment by 128-bit-guy** [comment]
The start is where cart enters warping tracks. The end is where cart would go if track was not warping with current switch states. For each cart each track kit and detector on path should be activated in one tick. This way does not allows to use any complex redstone based routing, but it allows to route tracks using tickets and routing books. (sorry for my bad english)

**Comment by liach** [comment]
Another issue is about your idea of having it powered in order to warp. This would not work because power status is reserved for track kits to perform their functionalities.

**Comment by 128-bit-guy** [comment]
Powering code may be exactly same as in electrical tracks

**Comment by liach** [comment]
Ah, you meant the energy instead of redstone. I thought you meant that the track is redstone activated

**Comment by WarlordWossman** [comment]
What is the point of playing with a mod about rails if you end up teleporting minecarts around?

@128-bit-guy if you don't want RC to become pointless with OP stuff from other mods you should probably disable easy creative fly and all the other teleporting features, they are rather boring anyways if you ask me and exists for people who would never use RC in the first place.

**Comment by liach** [comment]
This does have a point. It can allow you to test your signal and logic setups conveniently, though I agree that this track, if it exists, should have its crafting recipes disabled or limited to creative by default.

**Comment by WarlordWossman** [comment]
Ah okay, well that is another story, the initial issue sounded very much like it would be an item available in survival and that's what I mostly disagree with.
Not against handy creative tools tho.

**Comment by bookerthegeek** [comment]
What about a survival track that just teleports you cross dimension to the same cords? You could have one for the end and the nether.... Please? :)

**Comment by CovertJaguar** [comment]
Traversing the rails at warp speeds sounds like an implementation nightmare. And I also agree the gameplay aspect is counter productive.

I have a design drawn up for a more interesting system called Rifts. World gen bound gateways that you can power and redirect to a Rift in another dimension. The idea being that you still need to travel some distance to use a Rift. By chaining together multiple Rifts you can create an interesting rail network that spans an entire world, but still feels like a railway.

**Comment by bookerthegeek** [comment]
@CovertJaguar That would be amazing and I look forward to it. Keep up the good work!
